mod/groups/classes/Elgg/Upgrades/PluginTestUpgrade.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Elgg\Upgrades;
+
+/**
+ * This is an example of an upgrade that plugins can provide for Elgg's
+ * upgrading feature.
+ */
+class PluginTestUpgrade implements BatchUpgrade {
+
+	private $offset;
+
+	public function getTitle() {
+		return 'This upgrades stuff and things in groups';
+	}
+
+	public function getDescription() {
+		return 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam euismod auctor dolor, eget porta leo pretium in. Vivamus eget scelerisque urna. Sed pretium tellus quis risus tincidunt, id tincidunt elit ornare.';
+	}
+
+	public function isRequired() {
+		// This should be an actual database query or some
+		// check made against the dataroot
+		return true;
+	}
+
+	public function getTotal() {
+		return 80000;
+	}
+
+	public function run($offset = null) {
+		$this->offset = $offset;
```

Comment:
That shouldn't happen unless the ```getNextOffset()```has been implemented wrong. I don't think UpgradeService can verify that in any way. It just needs to trust the return value.